### DISCUSSION TITLE
About Stream with context not work on IIS

### DISCUSSION THREAD

**Discussion by dinhan92** [pallets/flask#5284] [Q&A]
On Local, I can make @stream_with_context work and generate a stream string to my mobile flutter app.
However when publish to IIS windows server in production, it does not work anymore, it return the whole string.

Is there a way to make it work in production with IIS?

My test code:

``` python
@app.route('/api/test', methods=['GET', 'POST'])
def test(**kwargs) -> str:
    try:
        test = "My test string."
        @stream_with_context
        def generate():
            for text in test:
                yield text;

        return app.response_class(generate())

```

**Comment by RonnyPfannschmidt** [comment]
This likely is a issue with configuring iis, as no details on the deployment configuration was given, there's nothing we could answer in good conscience

**Comment by dinhan92** [comment]
Here is my web config:

<configuration>
  <system.webServer>
    <handlers>
        <add name="Python FastCGI" path="*" verb="*"
modules="FastCgiModule"
scriptProcessor="...path\env\Scripts\python.exe|...path\env\Lib\site-packages\wfastcgi-3.0.0-py3.11.egg\wfastcgi.py"
resourceType="Unspecified" requireAccess="Script" />
    </handlers>
  </system.webServer>
  <appSettings>
    <add key="WSGI_HANDLER" value="TestAIChat.app" /> 
    <add key="PYTHONPATH" value="C:\inetpub\wwwroot\aichatpythonapi" />
<add key="WSGI_LOG" value="C:\inetpub\wwwroot\aichatpythonapi\app.log" />
  </appSettings>
</configuration>

When adding website, I just right click on Sites on IIS and choose Add
Websites.
I installed CGI
I followed this video to add website: https://www.youtube.com/watch?v=Cer6HwrVgyg

**Reply by RonnyPfannschmidt** [reply]
wfastcgi is unmaintained since more than half a decade, please check the current docs on iis and python

**Comment by dinhan92** [comment]
I'm new to python, can you recommend which one I should use besides
wfastcgi? I want to deploy my app with Windows server.
Thank you

**Reply by RonnyPfannschmidt** [reply]
i haven't used windows outside of family tech support in more than 15 years
